### PR TITLE
Feature2

### PR DESCRIPTION
**Pull request by shivraj-webonise**

### CODE REVIEW

File: blog/app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
     @post = current_user.posts.build
   end
 
-  def edit
-  end
+  def edit()end
 
   def create
-    @post = current_user.posts.build(post_params)
+    @post = current_user.posts.create(post_params)
```
Comment (eliza-abraham) [inline blog/app/controllers/posts_controller.rb]: do you need create and save?

File: blog/app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-  before_action :set_post, only: %i[ show edit update destroy ]
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
 
   def index
     if user_signed_in?
       @posts = current_user.posts
     else
-      @posts = Post.all
+      @posts = Post.where(:published => true)
```
Comment (eliza-abraham) [inline blog/app/controllers/posts_controller.rb]: use scope

File: blog/app/views/posts/edit.html.erb [hunk 3]
```diff
@@ -1,59 +1,49 @@
...
+        <%= link_to 'Show', @post , class: 'btn btn-primary', style: 'margin-top: 20px' %>    
+
+      </div>
     </div>
+  </div>
 </div>
 
 
+
+
```
Comment (eliza-abraham) [inline blog/app/views/posts/edit.html.erb]: This should have shown up on rubocop. New line at eof

File: blog/spec/models/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  end
+  it 'has a author' do
+    post = FactoryBot.create(:post)
+    expect(post).to be_valid
+  end
+  it 'has a content' do
+    post = FactoryBot.create(:post)
+    expect(post).to be_valid
+  end
+end
```
Comment (eliza-abraham) [inline blog/spec/models/post_spec.rb:22]: can you identify more test cases other than validation?

File: blog/app/controllers/posts_controller.rb (hunk 1 above)
Comment (shivraj-webonise) [reply blog/app/controllers/posts_controller.rb]: No actually create also by default saves it. I will change it to build

File: blog/app/controllers/posts_controller.rb (hunk 2 above)
Comment (shivraj-webonise) [reply blog/app/controllers/posts_controller.rb]: Okay I will use

File: blog/app/views/posts/edit.html.erb (hunk 3 above)
Comment (shivraj-webonise) [reply blog/app/views/posts/edit.html.erb]: I will correct it.

File: blog/spec/models/post_spec.rb (hunk 4 above)
Comment (shivraj-webonise) [reply blog/spec/models/post_spec.rb:22]: I will try and find out more test cases.

**Review by eliza-abraham — APPROVED** [review]

**Pull request merged**